lib/data/hash/map.ml

```diff
@@ -0,0 +1,133 @@
+open import "../../prelude.ml"
+open import "../hash.ml"
+include import "../index.ml"
+
+private module Math = import "../../lua/math.ml"
+private module Array = import "../array.ml"
+
+private type pairs 'k 'v =
+  | Nil
+  | Cons of { k : 'k, v : 'v, next : pairs 'k 'v }
+
+instance functor (pairs 'k) begin
+  let f <$> x =
+    match x with
```

Comment:
Sorry to nitpick like this, but do you think you could rewrite this with a local worker? To make the dictionary non-recursive